.github/workflows/release.yaml

```diff
@@ -0,0 +1,47 @@
+name: Build and Push Docker Image
+
+on:
+  release:
+    types: [published]  # Triggers on release tagging
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Log in to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
```

Comment:
can you explain what this is for?

Reply:
Docker uses a layered approach, building on top of base images etc. Locally, these layers all get cached. This layers are cached to speed up builds/image pulls etc. In github actions you need to use this action to preserve the caches between runs otherwise the cache is empty each time. Not strictly necessary but this was included in the suggestion workflow I found.  I believe we're already paying for it (and there's a certain amount free for public repos).